src/pages/products/[id].astro

```diff
@@ -24,7 +24,11 @@ const { product } = productId ? await getProduct(productId) : null;
         <ProductImage productImages={product.images} client:load />
 
         <!-- Product info -->
-        <ProductInfo productId={productId!} product={product} client:load />
+        {
+          productId && (
```

Comment:
using `&&` is a bad practice, always use ternary `product ? <products> : 'No product found`